scripts/common/rook.sh

```diff
@@ -246,3 +246,33 @@ function rook_operator_ready() {
     fi
     return 0
 }
+
+function rook_is_healthy_to_upgrade() {
+    log "Awaiting to check if rook-ceph ceph is health"
+    if ! $DIR/bin/kurl rook wait-for-health 600 ; then
+        kubectl -n rook-ceph exec deploy/rook-ceph-tools -- ceph status
+        logFail "Failed to detect Rook Ceph is not healthy"
+        return 1
+    fi
+
+    log "Checking Rook versions and replicas"
```

Comment:
I'm wondering if we should also check the stability of the Rook operator via the [`cephcluster`](https://rook.io/docs/rook/v1.10/CRDs/Cluster/ceph-cluster-crd/#status) CR. Namely, checking the `status.phase` field of the CR. Similar to what we do when we wait for Rook to [rollout successfully](https://github.com/replicatedhq/kURL/blob/f7833d757bf76ac2068311b2e8e9cf28cea79595/addons/rook/template/base/install.sh#L1077).

Reply:
If we run `kubectl -n rook-ceph describe cephcluster rook-ceph ` we have:

```
Status:
  Ceph:
    Health:        HEALTH_OK
    Last Checked:  2023-03-15T15:12:51Z
  State:           Created
Events:            <none>
```

With  `rook wait-for-health` we check the Ceph status (https://github.com/replicatedhq/kURL/blob/main/pkg/rook/health.go#L35-L64) using the toolbox : https://rook.io/docs/rook/v1.10/Troubleshooting/ceph-toolbox/?h=toolbox#interactive-toolbox (which provide further info)
 
Also, see that we have a logic to ignore some warnings because they does not affect the operations: https://github.com/replicatedhq/kURL/blob/main/pkg/rook/health.go#L69-L104 

Therefore, I think we should not check the status of cephcluster because:
- a) We check it directly using the toolbox already
- b) It might have warnings that are OK and expected to have which would cause the migration fail when it should not.

I hope that clarifies.